### DISCUSSION TITLE
Prompt for citation style with embark

### DISCUSSION THREAD

**Discussion by asymmetric** [emacs-citar/citar#823] [Q&A]
Typically, I can call `org-cite-insert` with `C-u` to be prompted about the citation style I intend to use. But sometimes I forget, and simply call `org-cite-insert`, only to later realize I'd like to use a non-default style.

Is it possible to have an embark action for the citar prompt, so that when I call `embark-act` from inside the citar minibuffer prompt, it would select the entry under the cursor, prompt me about a style, and then insert the citation in the buffer?

**Comment by bdarcus** [comment]
Just checking:

Are you aware if you run org-cite-insert again when in the left most part
of the citation, it will prompt you for the style?

**Reply by asymmetric** [reply]
Yes, I'm aware :)

But what I'm asking is for an embark action when I'm still in the citar interface, choosing a key. Partially because I often forget to use the universal prefix, partly because it's cumbersome to invoke.

**Reply by asymmetric** [reply]
I use what you mention, but after inserting a citation the cursor is at the end of it, so navigating to the beginning of it just to invoke `org-cite-insert` seems cumbersome as well, rather than calling this hypothetical embark action when i realize I've forgotten to use `C-u`, namely while I'm still in the citar UI.

**Comment by bdarcus** [comment]
Maybe you need an alternate command that is like the insert with prefix (like my-org-cite-insert-with-style), and then include that in the embark map?

**Reply by bdarcus** [reply]
Maybe this would help?

https://stackoverflow.com/questions/6156286/emacs-lisp-call-function-with-prefix-argument-programmatically

**Reply by asymmetric** [reply]
Thanks for the suggestion!

What I'm trying to do is slightly different. I basically understood embark as an ["Actually..."](https://karthinks.com/software/fifteen-ways-to-use-embark/#embark-act-actually-dot-dot-dot-dot-and-but-first-dot-dot-dot), so in my case, I called `M-x citar-open` but *actually* I should've called it with `M-x C-u citar-open`, so I'd like to use embark to allow me to add the `C-u` ex-post, so to speak.

Do you think what I'm asking is currently possible with some glue code in my personal config, or does it require some changes in citar itself?

**Reply by bdarcus** [reply]
I do understand; I just think you're asking a question about embark: is it possible to invoke a prefix argument on the `embark-act` selection?

The embark README seems to suggests the answer is yes:

>  if you are using the key bindings to call actions, you can pass prefix arguments to actions in the normal way. For example, you can use `C-u X2` with the above demonstration actions to make the message printed by example-action-command2 more emphatic

Have you tried that?

Above I was assuming you can't, so that my suggestion was writing a new command that does that for you. In that case, you'd use your own keymap for citar and embark, and include that command there.

Either way, no, I don't think this is anything to do with citar per se.

Make sense?
